infra/README.md

```diff
@@ -0,0 +1,130 @@
+This is a work in progress to use Terraform to manage our cloud
+infrastructure.
+
+Instructions:
+
+- You will need a GCP project.  Configure `gcloud` to point at your project:
+
+   ```
+   gcloud config set project <gcp-project-id>
+   gcloud config set compute/zone <gcp-zone>
+   ```
+
+- Create a service account for Terraform with Owner role, create a
+  service account key and place it in
+  `$HOME/.hail/terraform_sa_key.json`.
+
+- Enable the the GCP services needed by Hail:
+
+   ```
+   gcloud services enable \
+       compute.googleapis.com \
+       cloudresourcemanager.googleapis.com \
+       servicenetworking.googleapis.com \
+       sqladmin.googleapis.com \
+       container.googleapis.com \
+       serviceusage.googleapis.com
+   ```
+
+- Install terraform.
+
+- Create `$HOME/.hail/global.tfvars` that looks like:
+
+   ```
+   gcp_project = "<gcp-project-id>"
+   gcp_region = "<gcp-region>"
+   gcp_zone = "<gcp-zone>"
+   domain = "<domain>"
+   ```
+
+- Run `terraform init`.
+
+- Run `terraform apply -var-file="$HOME/.hail/global.tfvars"`.
+  Terraform has created a GKE cluster named `vdc`.  Configure `kubectl` to point at the vdc cluster:
+
+  ```
+  gcloud container clusters get-credentials vdc
+  ```
+
+- Go to the Google Cloud console, VPC networks > internal > Private
+  service connection > Private connections to services, and enable
+  Export custom routes to both connections.
+
+You can now install Hail:
+
+- Create a VM on the internal network, 100GB, Ubuntu 20.04 TLS, allow
+  full access to all Cloud APIs.  10GB will run out of space.  We
+  assume the rest of the commands are run on the VM.
+
+- Standardize file permissions.  This is for docker, which considers
+  permissions for caching.  Run `echo 'umask 022' > ~/.profile`.  You
+  will need to log out/in or run `umask 022`.
+
+- Update $HAIL/config.mk with your infrastructure settings.  You can
+  get settings from the default/global-config secret:
+
+  ```
+  kubectl get secret global-config -o json | jq '.data | map_values(@base64d)'
+  ```
+
+- Install some dependencies on the VM:
+
+  ```
+  sudo apt update
+  sudo apt install -y docker.io python3-pip
+  sudo usermod -a -G docker $USER
+  gcloud -q auth configure-docker
+  gcloud container clusters get-credentials --zone us-central1-a vdc
+  git clone https://github.com/cseed/hail.git
+  python3 -m pip install -r $HOME/hail/docker/requirements.txt
+  ```
+
+  You will have to log out/in for usermod to take effect.
+
+- Run `PROJECT=<gc-project-id>
+  $HAIL/docker/third-party/copy_images.sh`.  This copies some base
+  images from Dockerhub (which now has rate limits) to GCR.
+
+- Generate TLS certificates.  See ../dev-docs/tls-cookbook.md.
+
+- Run `kubectl -n default apply -f $HAIL/ci/bootstrap.yaml`.
+
+- Build the CI utils image.  Run `make push-ci-utils` in $HAIL/ci.
+
+- Deploy the bootstrap gateway.  Run `make deploy` in
+  $HAIL/bootstrap-gateway.
```

Comment:
This gateway backends will crash-loop, as the router service hasn't been deployed yet:
```
nginx: [emerg] host not found in upstream "router" in /etc/nginx/conf.d/gateway.conf:73
```

That's not a big deal, as the bootstrapping later will take care of it, but maybe worth mentioning.